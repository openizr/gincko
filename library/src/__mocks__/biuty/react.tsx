/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';

/**
 * biuty/react mock.
 */
function Component(props: JSXElement): JSX.Element {
  const { onChange, onFocus, onClick } = props;
  const { transform, onKeyDown, onBlur } = props;

  // Covers `onChange` handler.
  setTimeout(() => {
    onChange?.('test');
    if (process.env.IS_DATE === 'true') {
      onChange?.('2020/02/20');
    }
  }, 10);

  // Covers `onFocus` handler.
  onFocus?.();

  // Covers `onBlur` handler.
  onBlur?.();

  // Covers `transform` function.
  transform?.('ok');
  transform?.('1002');
  transform?.('100220');
  transform?.('10022020');

  // Covers `onKeyDown` handler.
  onKeyDown?.('', { key: '1' });
  onKeyDown?.('', { key: 'A', ctrlKey: false, preventDefault: vi.fn() });
  // Covers `onClick` handler.
  if (process.env.NOCLICK !== 'true') {
    setTimeout(() => {
      onClick?.();
    }, 10);
  }
  return (
    <div>{JSON.stringify(props)}</div>
  );
}

export const UILink = Component;
export const UIButton = Component;
export const UIOptions = Component;
export const UITextfield = Component;
export const UITextarea = Component;
export const UIFilePicker = Component;
export const buildClass = vi.fn((...values: string[]): string => values.join(' '));
export const markdown = vi.fn((label: string, lightMode: boolean) => `MARKDOWN FOR ${label}, ${lightMode}`);
