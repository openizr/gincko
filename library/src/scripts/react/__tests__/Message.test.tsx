/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment jsdom
 */

import React from 'react';
import Message from 'scripts/react/Message';
import { render } from '@testing-library/react';

jest.mock('biuty/react');

const JSXMessage = Message as JSXElement;

describe('react/Message', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - default props', async () => {
    const { container } = render(<JSXMessage id="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - custom props', async () => {
    const { container } = render(<JSXMessage id="test" label="*Test*" modifiers="strong" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
