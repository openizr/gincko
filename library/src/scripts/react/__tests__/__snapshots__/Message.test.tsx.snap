// Vitest Snapshot v1

exports[`react/Message > renders correctly - custom props 1`] = `
<section
  class="ui-message strong"
  id="test"
>
  MARKDOWN FOR *Test*, false
</section>
`;

exports[`react/Message > renders correctly - default props 1`] = `
<section
  class="ui-message "
  id="test"
>
  MARKDOWN FOR , false
</section>
`;
